src/main/scala/net/snowflake/spark/snowflake/Parameters.scala

```diff
@@ -305,81 +307,87 @@ object Parameters {
       case (key, value) => (key.toLowerCase, value)
     }
 
+    if (!userParameters.contains(PARAM_CONNECTION_ID)) {
```

Comment:
If connection id is presented in the parameters, all login related parameters will be optional.